src/main/java/com/yelp/nrtsearch/server/luceneserver/IndexState.java

```diff
@@ -641,6 +646,9 @@ public synchronized void load(JsonObject jsonObject) throws IOException {
   @Override
   public void close() throws IOException {
     logger.info(String.format("IndexState.close name= %s", name));
+    if (isPeriodicCommitEnabled()) {
```

Comment:
I would probably just do a null check, since that is what really matters.

Reply:
I think we need to do this since we can have the periodic commit disabled.